### ISSUE TITLE
Mac OSx unit tests failing on develop

### ISSUE THREAD

**Issue by alalazo** [spack/spack#7436]
The nightly build that includes Mac OS is failing on `develop` as #7314 removed the option to create unsigned binary packages with `-y`. An example of a failing build log is [here](https://travis-ci.org/spack/spack/jobs/350987182). For convenience, the relevant part of the stack trace is:
```console
<omitted>
       else:
            # create build cache without signing
            args = parser.parse_args(
>               ['create', '-d', mirror_path, '-y', str(spec)])
lib/spack/spack/test/packaging.py:160: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
lib/spack/external/argparse.py:1728: in parse_args
    self.error(msg % ' '.join(argv))
lib/spack/external/argparse.py:2392: in error
    self.exit(2, _('%s: error: %s\n') % (self.prog, message))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = ArgumentParser(prog='spack', usage=None, description=None, version=None, forma...lass=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True)
status = 2, message = 'spack: error: unrecognized arguments: -y\n'
    def exit(self, status=0, message=None):
        if message:
            self._print_message(message, _sys.stderr)
>       _sys.exit(status)
E       SystemExit: 2
lib/spack/external/argparse.py:2380: SystemExit
```

### Expected Result

A reassuring green light from Travis CI

### Actual Result

Mac OS tests failing, as one of the test was not updated (see above)

@gartung @scheibelp @tgamblin

**Comment by gartung** [comment]
@alalazo what is the command that is failing? 
I can put back the -y option

**Comment by gartung** [comment]
I see the problem now. I had gpg2 install on my macOS system when I ran the tests.

**Comment by gartung** [comment]
#7440 fixes this

**Comment by scheibelp** [comment]
This should now be resolved as #7440 is merged. Please reopen if the problem persists.
